Automattic/jetpack
# RNMobile: Ensure text is always visible within Contact Info block

Fixes an issue with the Contact Info block within the app, in which typed text can sometimes be invisible. 

The issue occurs when a block based theme is active with a background colour that contrasts with the device's light/dark mode. For example, if a theme's background is light while the device has dark mode enabled, the text will be white on white. 

Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6341

## Proposed changes:

* The logic associated with the block's text styles has been updated. If a block based theme is detected, its text colour will now be used to style text as its typed into the block.

### Other information:


## Jetpack product discussion

N/A

## Does this pull request change what data or activity we track or use?

No, it does not.

## Testing instructions:

Follow the _preparation steps_ once for each mode (dark/light) and then proceed with the _testing steps_.

### Preparation steps

* Ensure the device is in either dark mode or light mode.
* Activate a theme with support for global styles, e.g. [Byrne](PdyWFn-9-p2).
* In your site's /wp-admin area, navigate to the site editor (Appearance → Editor) and select a style corresponding to the opposite mode (dark background for light mode, and vice versa).
* In the app, navigate to the editor for any of the site's posts or pages.

### Testing steps

* Tap the + icon to open the block inserter and add a Contact Info block.
* Type text into each of the block's fields.
* Verify that you're able to see the text you're typing into the block's fields.

### Additional testing

* Ensure there are no regressions associated with non-block based themes i.e. dark mode should still retain dark styles and vice versa.

## Screenshots

<details>
 <summary>Dark mode + block based theme with light background ⤵️</summary>
 
| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/Automattic/jetpack/assets/2998162/ef43004d-d596-4a39-8c4e-2b34c61a7ab7" width="100%"> | <img src="https://github.com/Automattic/jetpack/assets/2998162/2db719cc-2196-437a-a35a-06dfcf438d6d" width="100%"> |
</details>
 
<details>
 <summary>Light mode + block based theme with dark background ⤵️</summary>
 
| Before  | After |
| ------------- | ------------- |
| <img src="https://github.com/Automattic/jetpack/assets/2998162/b0378803-addd-43ac-8650-c8210df2294a" width="100%"> | <img src="https://github.com/Automattic/jetpack/assets/2998162/398e6c18-6a60-44cd-bd4e-44c4e13fd556" width="100%"> |
</details>

<details>
 <summary>Non-block based themes ⤵️</summary>
 
| Dark  | Light |
| ------------- | ------------- |
| <img src="https://github.com/Automattic/jetpack/assets/2998162/f12752c3-5dbf-4f17-9f0b-085d9d7123a2" width="100%"> | <img src="https://github.com/Automattic/jetpack/assets/2998162/3a2b0df6-9bb4-4caa-a4d4-d22cb01a93d3" width="100%"> |
</details>
